tests/test_bpscore.py

```diff
@@ -0,0 +1,94 @@
+import os
+import shutil
+import tempfile
+import unittest
+from random import choice
+from string import ascii_letters
+import lsst.ctrl.bps.bps_core as bps
+
+
+class TestExecute(unittest.TestCase):
+
+    def setUp(self):
+        self.file = tempfile.NamedTemporaryFile("w+")
+
+    def tearDown(self):
+        self.file.close()
+
+    def testSuccessfulExecution(self):
+        """Test exit status if command succeeded."""
+        status = bps.execute('true', self.file.name)
+        self.assertIn('true', self.file.read())
+        self.assertEqual(status, 0)
+
+    def testFailingExecution(self):
+        """Test exit status if command failed."""
+        status = bps.execute('false', self.file.name)
+        self.assertIn('false', self.file.read())
+        self.assertNotEqual(status, 0)
+
+
+class TestPrettyPrinting(unittest.TestCase):
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def testTaskLabelFormatting(self):
+        """Test if a task label is properly formatted."""
+        original = "'raw+{band: i, instrument: HSC, detector: 17}'"
+        expected = "'raw\nband=i\n instrument=HSC\n detector=17'"
+        result = bps.pretty_dataset_label(original)
+        self.assertEqual(result, expected)
+
+
+class TestBpsCore(unittest.TestCase):
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def testCreatingQuantumGraph(self):
+        """Test if an appropriately named pickle file created."""
```

Comment:
Maybe expand this description to note that not testing integration with QuantumGraph generation code, but rather testing that BPS code executes command given in BPS config.